File: 본헌/BOJ_1874.py
N = int(input())
stack, ans, res, var = [], [], True, 1
for _ in range(N):
    num = int(input())
    while var <= num:
        stack.append(var)
        ans.append('+')
        var += 1
    if num == stack[-1]:
        stack.pop()
        ans.append('-')
    else:
        res = False
if not res:
    print('NO')
else:
    for i in ans:
        print(i)

File: 병준/BOJ_1874.java
package algorithm;
// https://hongcoding.tistory.com/39
import java.io.*;
import java.util.Stack;

public class Boj_1874 {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));
        Stack<Integer> S = new Stack<>();
        StringBuilder answer = new StringBuilder();
        int N = Integer.parseInt(reader.readLine());
        int checkNum = 0;
        boolean noYes = true;
        for (int i=0;i<N;i++){
            int Num = Integer.parseInt(reader.readLine());
            if (Num>checkNum) {
                for (int j = checkNum+1; j <= Num; j++) {
                    S.push(j);
                    answer.append("+\n");
                }
                checkNum = Num;
//                System.out.println(checkNum);
            } else if (S.peek()!=Num) {
                noYes = false;
            }
            S.pop();
            answer.append("-\n");
//            System.out.println(S);
        }
        if (noYes == true) {
            System.out.println(answer);
        }else {
            System.out.println("No");
        }
        reader.close();
        System.out.println();
    }
}
